Tasks/AzureAppServiceManage/appinsightsmanage.ts

```diff
@@ -0,0 +1,163 @@
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import path = require('path');
+
+var azureRESTUtils = require('azurerest-common/azurerestutility.js');
+var azureUtils = require('azurerest-common/utility.js');
+var extensionManage = require('./extensionmanage.js');
+
+var APPLICATION_INSIGHTS_EXTENSION_NAME = "Microsoft.ApplicationInsights.AzureWebSites";
+
+export class AppInsightsManage {
+
+    private endpoint;
+    private appInsightsResourceGroupName: string;
+    private appInsightsResourceName: string;
+    private webAppName: string;
+    private resourceGroupName: string;
+    private specifySlotFlag: boolean;
+    private slotName: string;
+
+    constructor(endpoint, appInsightsResourceGroupName, appInsightsResourceName, webAppName, resourceGroupName, specifySlotFlag, slotName) {
+        this.endpoint = endpoint;
+        this.appInsightsResourceGroupName = appInsightsResourceGroupName;
+        this.appInsightsResourceName = appInsightsResourceName;
+        this.webAppName = webAppName;
+        this.resourceGroupName = resourceGroupName;
+        this.specifySlotFlag = specifySlotFlag;
+        this.slotName = slotName;
+    }
+
+    public async configureAppInsights() {
+        this.resourceGroupName  = (this.specifySlotFlag ? this.resourceGroupName : await azureRESTUtils.getResourceGroupName(this.endpoint, this.webAppName));
+        var appServiceDetails = await azureRESTUtils.getAppServiceDetails(this.endpoint, this.resourceGroupName, this.webAppName, this.specifySlotFlag, this.slotName);
+        
+        if(appServiceDetails && appServiceDetails.kind && appServiceDetails.kind == "app,linux") {
+            throw new Error(tl.loc("ApplicationInsightsNotSupportedForLinuxApp", this.webAppName));
+        } else {
+            var publishingProfile = await azureRESTUtils.getAzureRMWebAppPublishProfile(this.endpoint, this.webAppName, this.resourceGroupName, this.specifySlotFlag, this.slotName);
+            await this.installApplicationInsightsExtension(publishingProfile);
+
+            var appInsightsResource = await this.getApplicationInsightResource();
+            if (appInsightsResource == null) {
+                throw new Error(tl.loc('UnableToGetAppInsightsResource', this.appInsightsResourceName));
+            }
+
+            appInsightsResource = await this.linkAppInsightsWithAppService(appInsightsResource);
+            await this.configureInstrumentationKey(appInsightsResource);
+            await this.configureAppServiceAlwaysOnProperty();
+            await this.configureAppInsightsWebTest(appInsightsResource, publishingProfile);
+            console.log(tl.loc("SuccessfullyConfiguredAppInsights"));
```

Comment:
Can you include following tests in a Webapp configured with AppInsights... 1) attempt deploy and re-deploy 2) attempt deployment with removing extra files option checked 3) attempt deployment with removing app data folder content (Will this impact App insight webjobs?)